hack/test-cmd.sh

```diff
@@ -238,6 +238,9 @@ fi
 PATH="${KUBE_OUTPUT_HOSTBIN}":$PATH
 
 kube::log::status "Checking kubectl version"
+
+# Note that this fails on a dirty commit.  To avoid that from happening, .gitignore
```

Comment:
I agree that we don't want to commit bindata, and adding it to .gitignore is correct, but I think this comment may be inaccurate.

I made my client dirty:
```console
$ git diff
diff --git a/README.md b/README.md
index 24a54aa..d8b2bc5 100644
--- a/README.md
+++ b/README.md
@@ -1,3 +1,5 @@
+a change
+
 # Kubernetes
 
 [![GoReportCard Widget]][GoReportCard] [![GoDoc Widget]][GoDoc] [![Travis Widget]][Travis] [![Coverage Status Widget]][Coverage Status]
$ git describe --long --tags --always --dirty --abbrev=14
v1.3.0-alpha.4-921-g06f742a5ea5605-dirty
```

but `hack/test-cmd.sh` still passed:
```console
$ PATH=third_party/etcd:$PATH hack/test-cmd.sh
...
+++ [0601 17:13:09] TEST PASSED
```

It'd be instructive to figure out what caused this test to fail during development. But as far as I know, `kubectl version` only fails if the client/server versions don't match, which would probably only occur if you did a few independent incomplete builds.